README
Eminence is an xbmc skin that seeks simplicity, functionality and elegance with a modern look. 

Creative Commons Non Commercial 3.0 License

All icon images from iconmonstr.com see website for license terms
Weather font from meteocons


changelog.txt
[B]1.2.6[/B]
updated language files from Transifex
Add menu button for videos library mouse
Fix artistslideshow start on home
Locksettings order

[B]1.2.5[/B]
Fix home menu visibility bug
Fix addon hub options
Update transifex

[B]1.2.4[/B]
Updates to viewtypes

[B]1.2.3[/B]
Fix MyWeather blank screen

[B]1.2.2[/B]
Remove strings.po~ file
Remove MyPVR.xml for Helix
Fix "Icons" non localised label in MyWeather.xml
Add Now Next Channel - epg views

[B]1.2.1[/B]
Startup video before login screen

[B]1.2.0[/B]
Startup video option
Fix AM PM clock issue
Seekbar progress cache

[B]1.1.9[/B]
Add more hub option presets
Add OSD PVR Guide
Highlight colour for pvr osd channels/guide
Fix hub option height
Fix preset list
Side label now shows playlist name

[B]1.1.8[/B]
Kaitoast changes
Fix lyrics window
Fix pvr hub back functions
Fix watched indicators vars
Music osd lower
Option for "classic" extendedinfo details
Group label pvr

[B]1.1.7[/B]
Fully Customisable Hubs for Movies, TvShows and Music (resets previous customisations)
Ability to change path for movies and tvshow hubs spotlight

[B]1.1.6[/B]
Transifex update
Fix some seekbar / info issues
Fix DialogSelect and FileBrowser need aspectratio keep
Add codec info to extendinfo
Fix back button in pvr
Add next and now recording info in header
Remove time info on osd for livetv (only finishes at)

[B]1.1.5[/B]
Weather Icon selection option
Skin init now after yes no dialog
Remove debug window
Fix music files hub issue

[B]1.1.4[/B]
Preliminary PVR setup
New weather icons
Fix hub label width

[B]1.1.3[/B]
Fix Trailers not opening fullscreen

[B]1.1.2[/B]
Alignment bug hunt begins
Custom_ Windows
MyWeather
Next Aired
Global Search
Favourites

[B]1.1.1[/B]
MediaFilter
MediaSource
NetworkSetup
Font remove unicode attrb
SmartPlaylistRule/Editor
VideoOSDSettings
ProfileSettings
All control id changes complete

[B]1.1.0[/B]
Helix/Kodi work begins
AddonBrowser
DialogAddonBrowser
DialogContentSettings
Keyboard Updates
Fix List View alignment
Fix Wide View alignment
Fix Poster View alignment
Fix Lovefilm View alignment
Fix Banner View alignment
Fix BigList alignment
Fix Infolist alignment
Fix Bannerlist alignment

[B]1.0.11[/B]
Dialog progress
Fixes to side header

[B]1.0.10[/B]
Update transifex
New Ok and YESNO Dialog
Fix FileManager header length
Match Sidelabel to posterborder color

[B]1.0.9[/B]
Fix undefined id

[B]1.0.8[/B]
weather changes
transifex update

[B]1.0.7[/B]
move right hub consolidation
fix header sub label overlap
fix alignment of labels on home
tweak opaque colours
separate out homebarfg colour

[B]1.0.6[/B]
remove ability to delete first tile (301) to prevent lock out
fix header gradient overlay for music vis on home
match tile colour defaults
fix side arrow for song info
transparent home colour scheme
specific info for tvshows and episodes
bump context menu zorder
fix big list overlap in tvshows

[B]1.0.5[/B]
updates to next aired window
updates to global search addon

[B]1.0.4[/B]
fix mumusicnav not having side arrow
consolidate side arrows into one def
fix weather arrows not changing view
fix weather not holding second view on mouse move
rudimentary script.skin.info.service support for set extendedplot
fixes to nextaired
auto download artwork downloader after update
remove list scroll animations

[B]1.0.3[/B]
fix overlapping list labels
center home list
HomeBarBG colour def
add labels to wall view episodes
added burnt orange, emerald and lilac
change VARLabel to use ListItem.Label for songs as well
 - songs view now inherits track naming template 
 - Settings > Music > File Lists > Track Naming Template

[B]1.0.2[/B]
fix video and year labels in vid info
fix not being able to set favourites as widgets

[B]1.0.1[/B]
new info list style as Media Info 3 fills spot better
option for 'Alt style' in side blade for old list style
header tile colour takes from TopBar
fix dialogfilestacking
fix set background image disabled when fanart disabled
change to no default overlay
tweak fade levels
remove Highlight-NF
subtitles sync and cc flags
subtitles file name for comparison

[B]1.0.0[/B]
fix livetv record not focusing onright

[B]0.9.59[/B]
DialogSongInfo functional
fix Network Add
big list play fix in song files
button menu localise reload skin (oops!)
add background image skin setting replacing silhouette
 - now can set both a fallback fanart and an overlay background
make script.favourites a dep

[B]0.9.58[/B]
remove Defaults.xml
add DialogSongInfo.xml and VideoOverlay.xml
fix ampersand
fix unmatch paras
fix undefined ids
tidy some unused includes

[B]0.9.57[/B]
fix osd button labels showing
fix onleft for pvr to always goto hub

[B]0.9.56[/B]
REPOSITORY INCLUSION FIXES
xbmc.gui bumped to 5.0.1
lang="en" in description and summary 
remove TODO
change case of Defaults.xml to defaults.xml
remove executable bit on font files
fixes to dialogcontentsettings
fixes to weather font colours
fixes to list info font colours
fixes to pvr lists font colours
fixes to file manager
replace "-" by "noop"
remove <label>-</label> references
resize _screenshots to 1280x720
replace & with &amp;
fix missing space before colordiffuse tags
remove additional "-" in comment lines
replace tabs with spaces
remove BOMs
change LICENSE files EOLs from CR/LF (win) to LF (unix)

[B]0.9.55[/B]
Preparation for repository inclusion
Removal of script into own addon
script.eminencefunctions now referenced rather than directly
Change to flat theme as default
Old theme become Glass
script.eminencefunctions now a dependency
Clean up of fanart fade settings

[B]0.9.54[/B]
Skin Settings descriptions

[B]0.9.53[/B]
Media Info view for Movies/TvShows
Fanart fade amount option

[B]0.9.52[/B]
Added trailer button to context menu
Added update library button to nav menu
Kiosk mode
Fixes to side menu for playlist
Match header tile colour to theme colour in skin extras
3d Mode selection in osd

[B]0.9.51[/B]
Added ability to customise global widget type
Fix silhouette not showing

extras/scripts/insertitem.py
import xbmc, sys

def movetile():
    label = xbmc.getInfoLabel("Skin.String(home."+str(i)+".label)")
    path = xbmc.getInfoLabel("Skin.String(home."+str(i)+".path)")
    icon = xbmc.getInfoLabel("Skin.String(home."+str(i)+".icon)")
    bgcolor = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgcolor)")
    bgwidget = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget)")
    bgwidgettype = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget.type)")
    enabled = xbmc.getInfoLabel("Skin.HasSetting(home."+str(i)+".enabled)")
    
    # Disable the tile
    xbmc.executebuiltin("Skin.Reset(home."+str(i)+".enabled)")
    
    # Only move info and enable tile if it was enabled before
    if str(enabled) == "True":
        xbmc.executebuiltin("Skin.SetBool(home."+str(o)+".enabled)")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".label,"+str(label)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".path,"+str(path)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".icon,"+str(icon)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgcolor,"+str(bgcolor)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget,"+str(bgwidget)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget.type,"+str(bgwidgettype)+")")

def removetile():
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".enabled)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".label)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".path)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".icon)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".bgwidget)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[1])+".bgwidget.type)")

if sys.argv[1]:
    insertpoint = str(sys.argv[1])
    insertpoint = insertpoint[-3:]
else:
    exit

# Do 329 to 320
if int(insertpoint) < 320:
    for i in reversed(range(320,330)):
        o = i + 1
        movetile()
else:
    for i in reversed(range(int(insertpoint),330)):
        o = i + 1
        movetile()
    removetile()
    exit

# Need a special one for 312
if int(insertpoint) < 313:
    i = 312
    o = 320
    movetile()
else:
    removetile()
    exit

# Do 300 to 311
for i in reversed(range(int(insertpoint),312)):
    o = i + 1
    movetile()
removetile()


extras/scripts/eminence_functions.py
import xbmc, sys

def movetile():
    label = xbmc.getInfoLabel("Skin.String(home."+str(i)+".label)")
    path = xbmc.getInfoLabel("Skin.String(home."+str(i)+".path)")
    icon = xbmc.getInfoLabel("Skin.String(home."+str(i)+".icon)")
    bgcolor = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgcolor)")
    bgwidget = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget)")
    bgwidgettype = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget.type)")
    enabled = xbmc.getInfoLabel("Skin.HasSetting(home."+str(i)+".enabled)")
    
    # Disable the tile
    xbmc.executebuiltin("Skin.Reset(home."+str(i)+".enabled)")
    
    # Only move info and enable tile if it was enabled before
    if str(enabled) == "True":
        xbmc.executebuiltin("Skin.SetBool(home."+str(o)+".enabled)")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".label,"+str(label)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".path,"+str(path)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".icon,"+str(icon)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgcolor,"+str(bgcolor)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget,"+str(bgwidget)+")")
        xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget.type,"+str(bgwidgettype)+")")

def removetile():
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".enabled)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".label)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".path)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".icon)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".bgwidget)")
    xbmc.executebuiltin("Skin.Reset("+str(sys.argv[2])+".bgwidget.type)")

if sys.argv[1] in ['inserttile']:
    if sys.argv[2]:
        insertpoint = str(sys.argv[2])
        insertpoint = insertpoint[-3:]
    else:
        exit

    # Do 329 to 320
    if int(insertpoint) < 320:
        for i in reversed(range(320,330)):
            o = i + 1
            movetile()
    else:
        for i in reversed(range(int(insertpoint),330)):
            o = i + 1
            movetile()
        removetile()
        exit

    # Need a special one for 312
    if int(insertpoint) < 313:
        i = 312
        o = 320
        movetile()
    else:
        removetile()
        exit

    # Do 300 to 311
    for i in reversed(range(int(insertpoint),312)):
        o = i + 1
        movetile()
    removetile()

if sys.argv[1] in ['swapitem']:
    i = str(sys.argv[2])
    i = i[-3:]
    o = str(sys.argv[3])
    o = o[-3:]
        
    label = xbmc.getInfoLabel("Skin.String(home."+str(i)+".label)")
    path = xbmc.getInfoLabel("Skin.String(home."+str(i)+".path)")
    icon = xbmc.getInfoLabel("Skin.String(home."+str(i)+".icon)")
    bgcolor = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgcolor)")
    bgwidget = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget)")
    bgwidgettype = xbmc.getInfoLabel("Skin.String(home."+str(i)+".bgwidget.type)")
    enabled = xbmc.getInfoLabel("Skin.HasSetting(home."+str(i)+".enabled)")
    
    olabel = xbmc.getInfoLabel("Skin.String(home."+str(o)+".label)")
    opath = xbmc.getInfoLabel("Skin.String(home."+str(o)+".path)")
    oicon = xbmc.getInfoLabel("Skin.String(home."+str(o)+".icon)")
    obgcolor = xbmc.getInfoLabel("Skin.String(home."+str(o)+".bgcolor)")
    obgwidget = xbmc.getInfoLabel("Skin.String(home."+str(o)+".bgwidget)")
    obgwidgettype = xbmc.getInfoLabel("Skin.String(home."+str(o)+".bgwidget.type)")
    oenabled = xbmc.getInfoLabel("Skin.HasSetting(home."+str(o)+".enabled)")
    
    xbmc.executebuiltin("Skin.SetBool(home."+str(i)+".enabled)")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".label,"+str(olabel)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".path,"+str(opath)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".icon,"+str(oicon)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".bgcolor,"+str(obgcolor)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".bgwidget,"+str(obgwidget)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(i)+".bgwidget.type,"+str(obgwidgettype)+")")
    
    xbmc.executebuiltin("Skin.SetBool(home."+str(o)+".enabled)")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".label,"+str(label)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".path,"+str(path)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".icon,"+str(icon)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgcolor,"+str(bgcolor)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget,"+str(bgwidget)+")")
    xbmc.executebuiltin("Skin.SetString(home."+str(o)+".bgwidget.type,"+str(bgwidgettype)+")")
